stages/org.osbuild.coreos.platforms

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/python3
+"""
+Process all necessary operations related to CoreOS
+platforms is crucial and specific to CoreOS. This step
+is essential for CoreOS exclusively.
+"""
+SCHEMA_2 = r"""
+"options": {
+  "additionalProperties": false,
+  "properties": {
+    "platform": {
+      "description": "The artifactory platform name",
+      "type": "string"
+    },
+    "platforms_json": {
+        "description": "The path where platforms.json is stored",
+        "type": "string"
+    },
+    "bootpath": {
+      "type": "string",
+      "description": "The mounted location of the boot filesystem tree where the BLS entries and grubconfig  will be under ./loader/entries/*.conf",
+      "pattern": "^(mount|tree):\/\/\/",
+      "examples": ["tree:///boot", "mount:///", "mount:///boot"],
+      "default": "tree:///boot"
+    },
+    "deployment": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["osname", "ref"],
+      "properties": {
+        "osname": {
+          "description": "Name of the stateroot to be used in the deployment",
+          "type": "string"
+        },
+        "ref": {
+          "description": "OStree ref to create and use for deployment",
+          "type": "string"
+        },
+        "serial": {
+          "description": "The deployment serial (usually '0')",
+          "type": "number",
+          "default": 0
+        }
+      }
+    }
+  }
+},
+"devices": {
+  "type": "object",
+  "additionalProperties": true
+},
+"mounts": {
+  "type": "array"
+}
+"""
+import glob
+import json
+import os
+import osbuild.api
+import shutil
+import sys
+
+from urllib.parse import urlparse
+from osbuild.util import ostree
+
+
+def copy_file(source_path, destination_path):
+    # Check if source file exists
+    if not os.path.isfile(source_path):
```

Comment:
(niptick) this is a TOCTOU issue, here it's probably not important but in general I would try to avoid checking for the existence and then copying, it's fine to fail in the copy step